### ISSUE TITLE
Django using ASCI codec instead of UTF8

### ISSUE THREAD

**Issue by opensemanticsearch** [opensemanticsearch/open-semantic-search-appliance#4]
Using special chars in thesaurus or search seams to work since locale of the Debian Linux system is now en_US.UTF-8 which can handle UTF-8 chars.

Indexing filenames including special chars via API fails because Python 2.x file system functions using ASCII as default encoding, to which filenames of the UTF8 environment can not be converted while running standard Python filesystem functions os.walk or os.path.

An upgrade to Python 3 (Debian package python3-django) which should be done to run on new Django versions could solve the issue, since Python 3 default encoding is UTF-8 instead of ASCII.

**Comment by mcr-ksh** [comment]
not sure if this is related but on a german dataset we get:

> Tue Jan 24 20:25:24.354143 2017] [:error] [pid 21341] [client 10.x:64667] PHP Notice:  Undefined index: NOT_path in /usr/share/solr-php-ui/index.php on line 619
> [Tue Jan 24 20:25:24.403934 2017] [:error] [pid 21341] [client 10.x:64667] PHP Fatal error:  Cannot access property started with '\\0' in /usr/share/solr-php-ui/Apache/Solr/Response.php on line 201
> 

$data cant be parsed via json_decode

`        
      /**
         * Parse the raw response into the parsed_data array for access
         *
         * @throws Apache_Solr_ParserException If the data could not be parsed
         */
        protected function _parseData()
        {
                //An alternative would be to use Zend_Json::decode(...)
                $data = json_decode($this->_response->getBody());

                // check that we receive a valid JSON response - we should never receive a null
                if ($data === null)
`

**Comment by opensemanticsearch** [comment]
Can this error be reproduced? And another search (without special chars) works?
A short time after booting the VM the search isn't working yet because Solr search server is not up yet.

Which version do you use? Maybe the system charset (command: locale) ASCI (C) yet or UTF-8 like in the new VM versions?

**Comment by mcr-ksh** [comment]
well, it basically renders everything unusable. so pretty much reproducible. UTF8 is default on the installation (Ubuntu 16.04). I just loaded a pile of documents and this is what happened. I downgraded php to 5.6 on, but the same result. Somehow this is related to the JSON parsing only as I fired a solr query and that one was fine. If I debug print the $data, copy&paste it to a besite, its fine but php cant handle it. Not sure why. (Btw, I checked everything that php is working with UTF8 etc)
